egui_plot/src/items/mod.rs

```diff
@@ -34,16 +34,20 @@ pub struct PlotItemBase {
     id: Id,
     highlight: bool,
     allow_hover: bool,
+    pub(crate) show_in_legend: bool,
 }
 
 impl PlotItemBase {
+    /// The given name will show up in the legend and serves to track the visibility of items.
+    /// Multiple items may share the same name, in which case they also share a legend entry.
```

Comment:
well that's not entirely true though, it's the id that tracks all this. It's just that by default the id is created from that name.

```suggestion
    /// The given name is used to create the item's id and will show up in the legend.
    ///
    /// Multiple items may share the same name (and id), in which case they also share a legend entry.
    /// The item's id is used to for visibility tracking and to refer to items in the plot's response.
```

should probably expand doc on the id setter as well.